### PR TITLE
Add "newcontainer" hyper command to "start" runtime subcommand

### PR DESCRIPTION
**Pull request by jcvenegas**
This PR adds newcontainer hyper request   in start command. It still need to tests using a real shim and using docker.

After send newcontainer hyper request, the "start" flow continues as usual.

### CODE REVIEW

File: src/proxy.c [hunk 1]
```diff
@@ -732,3 +804,132 @@ cc_proxy_hyper_pod_create (struct cc_oci_config *config)
...
+
+	// FIXME match with config or find a good default
+       json_object_set_boolean_member (newcontainer_payload, "initialize", false);
+
+       json_object_set_array_member (process, "args", args);
+       json_object_set_array_member (process, "envs", envs);
+       json_object_set_object_member (newcontainer_payload, "process", process);
+
+       if (! cc_proxy_run_hyper_cmd (config, "newcontainer", newcontainer_payload)) {
+               g_critical("failed to run pod create");
```
Comment (devimc) [inline src/proxy.c:922]: "pod create" ?  or "newcontainer"

Comment (jcvenegas) [reply src/proxy.c:922]: ups, copy paste :(

**Comment by grahamwhaley** [comment]
Just a note/nitpick @jcvenegas - some of the commit messages could do with a 'subsystem: ' start to their subject lines - I know @jodh-intel already has a bunch of cleaning up to do in his tree, so let's see if we can help him by not adding more :-)

And, in one subject s/dont/don't/

File: src/proxy.c [hunk 2]
```diff
@@ -732,3 +804,132 @@ cc_proxy_hyper_pod_create (struct cc_oci_config *config)
...
+       gboolean ret = false;
+
+       if (! (config && config->proxy)) {
+               goto out;
+       }
+
+       if (! cc_proxy_connect (config->proxy)) {
+               goto out;
+       }
+       if (! cc_proxy_attach (config->proxy, config->optarg_container_id)) {
```
Comment (amshinde) [inline src/proxy.c:833]: You will need the attachIO proxy command instead for the create-start flow.

File: src/proxy.c [hunk 3]
```diff
@@ -732,3 +804,132 @@ cc_proxy_hyper_pod_create (struct cc_oci_config *config)
...
+       */
+       json_object_set_string_member (newcontainer_payload, "image", "");
+
+       /* newcontainer.process */
+       json_object_set_boolean_member(process, "terminal", true);
+
+       /* FIXME how to handle IO streams id's  ?*/
+       json_object_set_int_member (process, "stdio", 1);
+       json_object_set_int_member (process, "stderr", 2);
+
```
Comment (amshinde) [inline src/proxy.c:885]: Need to replace these with the seq numbers received from the attachIO command.

File: src/proxy.c [hunk 4]
```diff
@@ -732,3 +804,132 @@ cc_proxy_hyper_pod_create (struct cc_oci_config *config)
...
+#else
+	JsonObject *env_var = json_object_new ();
+	json_object_set_string_member (env_var, "env", "PATH");
+	json_object_set_string_member (env_var, "value", "/bin:/usr/bin");
+	json_array_add_object_element (envs, env_var);
+#endif
+
+	// FIXME use config cwd
+       json_object_set_string_member (process, "workdir", "/");
+
```
Comment (amshinde) [inline src/proxy.c:910]: I had pushed a patch to set the home dir to the user's home dir. You may want to use that. Thats how runc sets its home directory.

File: src/proxy.c [hunk 5]
```diff
@@ -732,3 +804,132 @@ cc_proxy_hyper_pod_create (struct cc_oci_config *config)
...
+        * FIXME ADD rootfs
+       */
+       json_object_set_string_member (newcontainer_payload, "rootfs", "");
+       /*
+        * FIXME ADD image
+       */
+       json_object_set_string_member (newcontainer_payload, "image", "");
+
+       /* newcontainer.process */
+       json_object_set_boolean_member(process, "terminal", true);
```
Comment (amshinde) [inline src/proxy.c:880]: The terminal value present in config.json should be used here.

**Comment by jcvenegas** [comment]
Note: Adding process from config.json into the state file will would break the start flow when is needed to wait or not (detach/attach).

https://github.com/01org/cc-oci-runtime/pull/336 fix the problem

**Pull request closed without merging**